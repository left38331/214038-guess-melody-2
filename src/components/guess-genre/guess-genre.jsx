import React from 'react';
import PropTypes from 'prop-types';
import {AudioPlayer} from 'components/audio-player/audio-player';

export class GuessGenre extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      activePlayer: null
    };
  }

  render() {
    return <section className="game game--genre">
      <header className="game__header">
        <a className="game__back" href="#">
          <span className="visually-hidden">Сыграть ещё раз</span>
          <img className="game__logo" src="img/melody-logo-ginger.png" alt="Угадай мелодию" />
        </a>

        <svg xmlns="http://www.w3.org/2000/svg" className="timer" viewBox="0 0 780 780">
          <circle className="timer__line" cx="390" cy="390" r="370" style={{filter: `url(#blur)`, transform: `rotate(-90deg) scaleY(-1)`, transformOrigin: `center`}}/>
        </svg>

        <div className="timer__value" xmlns="http://www.w3.org/1999/xhtml">
          <span className="timer__mins">05</span>
          <span className="timer__dots">:</span>
          <span className="timer__secs">00</span>
        </div>

        <div className="game__mistakes">
          <div className="wrong"></div>
          <div className="wrong"></div>
          <div className="wrong"></div>
        </div>
      </header>

      <section className="game__screen">
        <h2 className="game__title">Выберите инди-рок треки</h2>
        <form className="game__tracks" onSubmit={this.props.formSubmitHandler}>

          {this.props.question.answers.map((it, i) => {
            return <div className="track" key={it.id}>
              <AudioPlayer
                isPlaying={i === this.state.activePlayer}
                src={it.src}
                onPlayButtonClick={() => this.setState({
                  activePlayer: this.state.activePlayer === i ? null : i
                })}
              />
              <div className="game__answer">
                <input className="game__input visually-hidden" type="checkbox" name="answer" value={`answer-${i + 1}`} id={it.id} onChange={this.props.getValueForAnswer} />
                <label className="game__check" htmlFor={it.id}>Отметить</label>
              </div>
            </div>;
          })}

          <button className="game__submit button" type="submit">Ответить</button>
        </form>
      </section>
    </section>;
  }
}

GuessGenre.propTypes = {
  question: PropTypes.shape({
    type: PropTypes.string.isRequired,
    answers: PropTypes.arrayOf(PropTypes.object),
    genre: PropTypes.string.isRequired
  }),
  formSubmitHandler: PropTypes.func.isRequired,
  getValueForAnswer: PropTypes.func.isRequired
};
