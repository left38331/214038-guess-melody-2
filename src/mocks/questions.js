const questions = [
  {
    type: `artist`,
    song: {
      artist: `John Snow`,
      src: `https://upload.wikimedia.org/wikipedia/en/transcoded/7/7d/ACDC_Itsalongway.ogg/ACDC_Itsalongway.ogg.mp3`,
    },
    answers: [
      {
        id: 1,
        picture: `http://placehold.it/134x134`,
        artist: `John Snow`,
      },
      {
        id: 2,
        picture: `http://placehold.it/134x134`,
        artist: `Jack Daniels`,
      },
      {
        id: 3,
        picture: `http://placehold.it/134x134`,
        artist: `Jim Beam`,
      },
    ],
  },
  {
    type: `genre`,
    genre: `rock`,
    answers: [
      {
        id: 1,
        src: `https://upload.wikimedia.org/wikipedia/en/transcoded/7/7d/ACDC_Itsalongway.ogg/ACDC_Itsalongway.ogg.mp3`,
        genre: `rock`,
      },
      {
        id: 2,
        src: `https://upload.wikimedia.org/wikipedia/en/transcoded/8/89/My_Kind_Of_Town_clip.ogg/My_Kind_Of_Town_clip.ogg.mp3`,
        genre: `pop`,
      },
      {
        id: 3,
        src: `https://upload.wikimedia.org/wikipedia/en/transcoded/4/40/Madonna_-_You%27ll_See.ogg/Madonna_-_You%27ll_See.ogg.mp3`,
        genre: `jazz`,
      },
      {
        id: 4,
        src: `https://upload.wikimedia.org/wikipedia/ru/transcoded/d/d1/Rammstein_-_Keine_Lust.ogg/Rammstein_-_Keine_Lust.ogg.mp3`,
        genre: `rock`,
      },
    ],
  },
  {
    type: `artist`,
    song: {
      artist: `John Snow`,
      src: `https://upload.wikimedia.org/wikipedia/en/transcoded/7/7d/ACDC_Itsalongway.ogg/ACDC_Itsalongway.ogg.mp3`,
    },
    answers: [
      {
        id: 1,
        picture: `http://placehold.it/134x134`,
        artist: `John Snow`,
      },
      {
        id: 2,
        picture: `http://placehold.it/134x134`,
        artist: `Jack Daniels`,
      },
      {
        id: 3,
        picture: `http://placehold.it/134x134`,
        artist: `Jim Beam`,
      },
    ],
  },
  {
    type: `genre`,
    genre: `rock`,
    answers: [
      {
        id: 1,
        src: `https://upload.wikimedia.org/wikipedia/en/transcoded/7/7d/ACDC_Itsalongway.ogg/ACDC_Itsalongway.ogg.mp3`,
        genre: `rock`,
      },
      {
        id: 2,
        src: `https://upload.wikimedia.org/wikipedia/en/transcoded/8/89/My_Kind_Of_Town_clip.ogg/My_Kind_Of_Town_clip.ogg.mp3`,
        genre: `pop`,
      },
      {
        id: 3,
        src: `https://upload.wikimedia.org/wikipedia/en/transcoded/4/40/Madonna_-_You%27ll_See.ogg/Madonna_-_You%27ll_See.ogg.mp3`,
        genre: `jazz`,
      },
      {
        id: 4,
        src: `https://upload.wikimedia.org/wikipedia/ru/transcoded/d/d1/Rammstein_-_Keine_Lust.ogg/Rammstein_-_Keine_Lust.ogg.mp3`,
        genre: `rock`,
      },
    ],
  },
  {
    type: `artist`,
    song: {
      artist: `John Snow`,
      src: `https://upload.wikimedia.org/wikipedia/en/transcoded/7/7d/ACDC_Itsalongway.ogg/ACDC_Itsalongway.ogg.mp3`,
    },
    answers: [
      {
        id: 1,
        picture: `http://placehold.it/134x134`,
        artist: `John Snow`,
      },
      {
        id: 2,
        picture: `http://placehold.it/134x134`,
        artist: `Jack Daniels`,
      },
      {
        id: 3,
        picture: `http://placehold.it/134x134`,
        artist: `Jim Beam`,
      },
    ],
  },
  {
    type: `genre`,
    genre: `rock`,
    answers: [
      {
        id: 1,
        src: `https://upload.wikimedia.org/wikipedia/en/transcoded/7/7d/ACDC_Itsalongway.ogg/ACDC_Itsalongway.ogg.mp3`,
        genre: `rock`,
      },
      {
        id: 2,
        src: `https://upload.wikimedia.org/wikipedia/en/transcoded/8/89/My_Kind_Of_Town_clip.ogg/My_Kind_Of_Town_clip.ogg.mp3`,
        genre: `pop`,
      },
      {
        id: 3,
        src: `https://upload.wikimedia.org/wikipedia/en/transcoded/4/40/Madonna_-_You%27ll_See.ogg/Madonna_-_You%27ll_See.ogg.mp3`,
        genre: `jazz`,
      },
      {
        id: 4,
        src: `https://upload.wikimedia.org/wikipedia/ru/transcoded/d/d1/Rammstein_-_Keine_Lust.ogg/Rammstein_-_Keine_Lust.ogg.mp3`,
        genre: `rock`,
      },
    ],
  }
];

const settings = {
  gameTime: 300,
  errorCount: 3
};

export {questions, settings};
