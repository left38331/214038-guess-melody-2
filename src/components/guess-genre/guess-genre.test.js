import React from 'react';
import renderer from 'react-test-renderer';
import {GuessGenre} from 'components/guess-genre/guess-genre';
import {questions} from '../../mocks/questions';

const question = questions[1];

it(`render correctly guess genre component`, () => {
  const props = {
    formSubmitHandler: () => {},
    getValueForAnswer: () => {},
    question
  };

  const guessGenreComponent = renderer.create(<GuessGenre {...props} />, {
    createNodeMock: (element) => {
      if (element.type === `audio`) {
        return {};
      }
      return null;
    }}).toJSON();

  expect(guessGenreComponent).toMatchSnapshot();
});
