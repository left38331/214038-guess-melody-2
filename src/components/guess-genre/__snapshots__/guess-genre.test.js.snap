// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly guess genre component 1`] = `
<section
  className="game game--genre"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Выберите инди-рок треки
    </h2>
    <form
      className="game__tracks"
      onSubmit={[Function]}
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://upload.wikimedia.org/wikipedia/en/transcoded/7/7d/ACDC_Itsalongway.ogg/ACDC_Itsalongway.ogg.mp3"
          />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id={1}
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="answer-1"
          />
          <label
            className="game__check"
            htmlFor={1}
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://upload.wikimedia.org/wikipedia/en/transcoded/8/89/My_Kind_Of_Town_clip.ogg/My_Kind_Of_Town_clip.ogg.mp3"
          />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id={2}
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="answer-2"
          />
          <label
            className="game__check"
            htmlFor={2}
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://upload.wikimedia.org/wikipedia/en/transcoded/4/40/Madonna_-_You%27ll_See.ogg/Madonna_-_You%27ll_See.ogg.mp3"
          />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id={3}
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="answer-3"
          />
          <label
            className="game__check"
            htmlFor={3}
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://upload.wikimedia.org/wikipedia/ru/transcoded/d/d1/Rammstein_-_Keine_Lust.ogg/Rammstein_-_Keine_Lust.ogg.mp3"
          />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id={4}
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="answer-4"
          />
          <label
            className="game__check"
            htmlFor={4}
          >
            Отметить
          </label>
        </div>
      </div>
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;
